pkg/kubelet/kubelet_pods.go

```diff
@@ -435,6 +435,21 @@ func (kl *Kubelet) GetPodCgroupParent(pod *v1.Pod) string {
 	return cgroupParent
 }
 
+// GetHostNameForKernel gets hostname value to set in the hostname field (the nodename field of struct utsname) of the pod.
+func (kl *Kubelet) GetHostNameForKernel(hostname string, hostDomainName string, hostnameFQDN bool) (string, error) {
+	kernelhostname := hostname
+	if len(hostDomainName) > 0 && hostnameFQDN {
+		fqdnHostname := fmt.Sprintf("%s.%s", hostname, hostDomainName)
+		// FQDN has to be 63 chars to fit in hostname (specification is 64bytes which is 63 chars and the null terminating char).
```

Comment:
Where is the spec, kernel source points to 64+1 or 8+1
https://elixir.bootlin.com/linux/latest/source/include/uapi/linux/utsname.h#L5

Reply:
Good call, I think you are right. Each element of DNS has to be 63, but the utsname can be 64, so in my case the fqdn can be up to 64. I will update.